### ISSUE TITLE
ValidClassName.NotCamelCaps fails for anon classes

### ISSUE THREAD

**Issue by JeroenDeDauw** [squizlabs/PHP_CodeSniffer#956]
```php
$donation = $this->newDonationWithPaymentMethod( new class implements PaymentMethod {
	public function getType(): string {
		return '';
	}
} );
```

```
 123 | ERROR   | Class name "implements" is not in camel caps format
     |         | (Squiz.Classes.ValidClassName.NotCamelCaps)

```

> PHP_CodeSniffer version 1.5.5 (stable) by Squiz (http://www.squiz.net)

**Comment by gsherwood** [comment]
You're on an old unsupported branch of PHP_CodeSniffer so you don't have any of the PHP7-specific features, including anon class support.

To get support for anon classes, you'll need to upgrade to at least PHP_CodeSniffer 2.4.0 (released Nov 2015) although I'd strongly recommend moving to the latest 2.6.0.

I've tested this code on both 1.5.5 and 2.6.0 and can confirm the issue exists in 1.5.5 and not in 2.6.0. So closing this issue as it has been fixed in the 2.x branch.
